module_auto_update/models/module.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from odoo import api, fields, models
+from odoo.modules.module import get_module_path
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+
+class Module(models.Model):
+    _inherit = 'ir.module.module'
+
+    checksum_installed = fields.Char(
+        string="Installed Module Checksum",
+    )
+    checksum_dir = fields.Char(
+        string="Current Module Checksum",
+        compute='_compute_checksum_dir',
+    )
+    upgrade_available = fields.Boolean(
+        default=False,
+    )
+
+    @api.multi
+    def _compute_checksum_dir(self):
+        for r in self:
+            r.checksum_dir = dirhash(get_module_path(r.name), 'sha1')
```

Comment:
One question... If we would have addons directory full of `.pyc` or `.pyo` files, and under a containerized frozen environment where those files are autoregenerated on each build, would that produce a different checksum? I guess it should be the same because if the pycompiled file is based on the same input python file, it should produce the same output, but did you check this? Or is it possible to ignore such files?